code/modules/mob/living/simple_animal/simple_animal.dm

```diff
@@ -768,11 +768,10 @@ var/global/list/animal_count = list() //Stores types, and amount of animals of t
 				adjustBruteLoss(volume * 0.5)
 
 /mob/living/simple_animal/proc/delayedRegen()
-	set waitfor = 0
```

Comment:
`set waitfor = 0` causes the proc to essentially `spawn()` off as soon as a sleep is encountered. Changing this to use an actual `spawn()` directly does nothing (except slightly make performance worse).